app/gui/src/project-view/components/MarkdownEditor/MarkdownEditorImpl.vue

```diff
@@ -1,80 +1,137 @@
 <script setup lang="ts">
-import FloatingSelectionMenu from '@/components/FloatingSelectionMenu.vue'
-import FormattingToolbar from '@/components/MarkdownEditor/FormattingToolbar.vue'
-import { imagePlugin } from '@/components/MarkdownEditor/ImagePlugin'
-import SelectionFormattingToolbar from '@/components/MarkdownEditor/SelectionFormattingToolbar.vue'
-import { lexicalRichTextTheme, useFormatting } from '@/components/MarkdownEditor/formatting'
+import EditorRoot from '@/components/MarkdownEditor/EditorRoot.vue'
+import { highlightStyle } from '@/components/MarkdownEditor/highlight'
 import {
-  provideLexicalImageUrlTransformer,
-  type UrlTransformer,
+  provideDocumentationImageUrlTransformer,
+  UrlTransformer,
 } from '@/components/MarkdownEditor/imageUrlTransformer'
-import { listPlugin } from '@/components/MarkdownEditor/listPlugin'
-import { markdownPlugin } from '@/components/MarkdownEditor/markdown'
-import { useLexical } from '@/components/lexical'
-import LexicalContent from '@/components/lexical/LexicalContent.vue'
-import LexicalDecorators from '@/components/lexical/LexicalDecorators.vue'
-import { autoLinkPlugin, linkPlugin, useLinkNode } from '@/components/lexical/LinkPlugin'
-import LinkToolbar from '@/components/lexical/LinkToolbar.vue'
-import { shallowRef, toRef, useCssModule, type ComponentInstance } from 'vue'
+import { ensoMarkdown } from '@/components/MarkdownEditor/markdown'
+import VueComponentHost from '@/components/VueComponentHost.vue'
+import { EditorState } from '@codemirror/state'
+import { EditorView } from '@codemirror/view'
+import { minimalSetup } from 'codemirror'
+import { ComponentInstance, onMounted, ref, toRef, useCssModule, watchEffect } from 'vue'
+import { yCollab } from 'y-codemirror.next'
+import * as awarenessProtocol from 'y-protocols/awareness.js'
+import * as Y from 'yjs'
+
+const editorRoot = ref<ComponentInstance<typeof EditorRoot>>()
 
-const markdown = defineModel<string>({ required: true })
 const props = defineProps<{
+  yText: Y.Text
```

Comment:
Having `Y.Text` as a prop makes it deeply reactive. Now, to some extent this is desirable (so we don't need to listen to Ydoc updates), but I wonder if there are any guarantees what's inside Y.Text object? I'd like to avoid a situation when this property is updated because something changed in a "global" Ydoc structure, which is back referenced here.

Or prehaps it's enough to have it shallow-reactive? I don't know how yCollab exactly works.

Reply:
Good catch--yCollab listens for Yjs updates, so we don't need deep reactivity here. Refactored this component to watch the `yText` shallowly.

Generally speaking, Yjs objects are reasonably well-behaved if deep reactivity is applied to them. When a Yjs object is watched deeply it shouldn't cause *completely* spurious invalidation due to activity elsewhere in the document, but some more local spurious invalidation can occur due to mutation of objects that are adjacent in Yjs's internal doubly-linked list. The main reason it's more efficient to use them non-reactively and listen to Yjs updates when possible, is that we can make smaller changes when we receive a Yjs delta than when Vue reactivity tells us we need to recompute something.